docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,385 @@
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `http.error.reason` | string | Request failure reason. [1] | `timeout`; `name_resolution_error`; `secure_connection_error` | Conditionally Required: If request has failed. |
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [2] | `3.1.1` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Logical server hostname, matches server FQDN if available, and IP or socket address if FQDN is not known. | `example.com` | Recommended |
+| [`server.port`](../general/attributes.md) | int | Logical server port number | `80`; `8080`; `443` | Recommended |
+| [`server.socket.address`](../general/attributes.md) | string | Physical server IP address or Unix socket address. If set from the client, should simply use the socket's peer address, and not attempt to find any actual server IP (i.e., if set from client, this may represent some proxy server instead of the logical server). | `10.5.3.2` | Recommended: If different than `server.address`. |
+| [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `https`; `ftp`; `telnet` | Recommended |
+
+**[1]:** **TODO: there is a [open discussion/proposal in otel for http error code](https://github.com/open-telemetry/semantic-conventions/pull/205)**
+Provides low-cardinality error details: such as `timeout`, `cancellation` or one of [HTTP Request error](https://github.com/dotnet/runtime/blob/c430570a01c103bc7f117be573f37d8ce8a129b8/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs) in snake_case.
+
+**[2]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+`http.error.reason` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `` | No error. Default value. |
+| `_OTHER` | unknown error |
+<!-- endsemconv -->
+
+### Metric: `http.client.duration`
+
+<!-- semconv metric.dotnet.http.client.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.duration` | Histogram | `s` | Measures the duration of outbound HTTP requests. |
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.client.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `http.error.reason` | string | Request failure reason. [1] | `timeout`; `name_resolution_error`; `secure_connection_error` | Conditionally Required: If request has failed. |
```

Comment:
Given that this page is dotnet-http-metrics, would it make sense for the examples or other documentation to discuss the particular enumeration values that HttpClient would put in here?

Reply:
There is a note [1] on line 110 that talks about possible values.

At the moment yaml schema tooling does  not  support extending enums - https://github.com/open-telemetry/build-tools/issues/192 and we can't formally list all possible values.

There is also another note on line 131 that lists possible (common) error codes - once the above issue is resolved, custom values should be populated there.

Reply:
I moved the note to brief so that it appears in the table. LMK if it helps.